arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/cancelBoundary.kt

```diff
@@ -6,6 +6,28 @@ import kotlin.coroutines.intrinsics.suspendCoroutineUninterceptedOrReturn
 /**
  * Checks for cancellation,
  * when cancellation occurs this coroutine will suspend indefinitely and never continue.
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.coroutines.*
+ *
+ * //sampleStart
+ * suspend fun forever(): Unit {
+ *   while(true) {
+ *     println("I am getting dizzy...")
+ *     cancelBoundary() // cancellable computation loop
+ *   }
+ * }
+ *
+ * suspend fun main(): Unit {
+ *   val fiber = ForkConnected {
```

Comment:
I'm not clear about the name ForkConnected. Could you elaborate a bit about why the Connected part?

Reply:
I added docs for `ForkConnected` with an example after this comment :D [link](https://github.com/arrow-kt/arrow-fx/pull/248/files#diff-5b3bc4412001765ff242219bda005d13R41)

##### Reasoning for new `ForkConnected` and `ForkScoped` functions 

In Arrow Fx IO we only had `fork` which launches a `Fiber` without any cancellation connection control. Meaning you *always* leak unless you launch a `Fiber` inside `Resource` or `bracketCase` as the `acquire` step.
This makes `fork` quite complicated and easily results in cancellation bugs etc. Those semantics of IO's `fork` is now an combinator called `ForkAndForget.

Some more info [here](https://github.com/arrow-kt/arrow-fx/tree/master/arrow-fx-coroutines#racepairracetripleforkandforget). We should probably also document these, but this comes back to trying to figure out a way to add higher-level documentation to top-level functions besides function docs. Or do you think we should include all of that there?